btk/metrics.py

```diff
@@ -1,484 +1,110 @@
-    if segmentation is None:
-        return None
-    return None
-
-
-def evaluate_flux(flux, data=None, index=None):
-    if flux is None:
-        return None
-    return None
-
-
-def evaluate_shapes(shapes, data=None, index=None):
-    if shapes is None:
-        return None
-    return None
-
-
-def run(metrics_params, test_size=1000):
-    """Runs detection/segmentation/flux/shape measurement algorithm defined in
-    the input metrics params for input test_size number of btk runs.
-
-    Args:
-        metrics_params: Instance from class
-        `btk.metrics.Metrics_params` describing functions to return
-        results of detection/deblending/measurement algorithm.
-        test_size(int): Number of times Metrics_params is run and results
-            summarized.
-
-    Returns:
-        dict summarizing detection/deblending/measurement results.
 
-    """
-    results = {
-        "detection": [astropy.table.Table(), astropy.table.Table(), []],
-        "segmentation": [],
-        "flux": [],
-        "shapes": [],
-    }
-    for i in range(test_size):
-        print(f"Running test {i}")
-        # Evaluate detection algorithm
-        try:
-            batch_detection_result = metrics_params.get_detections()
-        except GeneratorExit as e:
-            print(e)
-            print("GeneratorExit encountered. Returning results")
-            return results
-        if (
-            len(batch_detection_result[0]) != len(batch_detection_result[1])
-            or len(batch_detection_result[0]) != metrics_params.batch_size
-        ):
-            raise ValueError(
-                "Metrics_params.get_detections output must be "
-                "two lists of astropy table of length batch size."
-                f" Found {len(batch_detection_result[0])}, "
-                f"{len(batch_detection_result[1])}, "
-                f"{metrics_params.batch_size}"
-            )
-        true_table, detected_table, detection_summary = evaluate_detection(
-            batch_detection_result[0], batch_detection_result[1], batch_index=i
-        )
-        results["detection"][0] = astropy.table.vstack([results["detection"][0], true_table])
-        results["detection"][1] = astropy.table.vstack([results["detection"][1], detected_table])
-        results["detection"][2].extend(detection_summary)
-        # Evaluate segmentation algorithm
-        segmentation = metrics_params.get_segmentation()
-        results["segmentation"].append(evaluate_segmentation(segmentation, index=i))
-        # Evaluate flux measurement algorithm
-        flux = metrics_params.get_flux()
-        results["flux"].append(evaluate_flux(flux, index=i))
-        # Evaluate shape measurement algorithm
-        shapes = metrics_params.get_shapes()
-        results["shapes"].append(evaluate_shapes(shapes, index=i))
-    return results
+def compute_metrics_wrap(
+    blend_results, measure_results, use_metrics=("detection", "segmentation", "reconstruction")
+):
```

Comment:
This made me think about something @thuiop in order to simplify the UI - maybe we can keep the structure of a generator for the metrics so introduce a `MetricsGenerator` that processes one batch from the `MeasureGenerator` at a time (I think it's effectively what you have so far in the code?). 

I feel like from the user perspective, they probably don't want things in batches for analyzing the results after everything is processed. So in addition, we can provide an utility function that takes in a `metrics_generator` which processes some specified # of batches, calculates metrics for all of them, and summarizes the results into 3 simple tables. I can imagine the 3 tables being: 

- `true_table`: Contains true properties from all galaxies, in all blends. Each row is indexed by 'blend_id' and 'gal_id' (the index of the blend, and the index of each galaxy in that blend). 

- `measure_table`: Same as `true_table` but containing `match_id` (-1 if not matched) and only the info implemented in the `measure_function` via the `catalog` key. 

- `metrics_table`: Each row is indexed by `blend_id` and contains the metrics on that blend (# gals in that blend, # matches, mse, etc.) 

Some more global metrics that need to be accumulated independently can be returned additionally as a dictionary. What do you think @thuiop ?

Reply:
We definitely need to think about the exact input and output format. Here my goal would be for users to have the data in the right format for the compute_metrics function and use it directly ; BTK users can use the "wrap" one directly, or we may have a generator which does that.
As for the output, we will definitely need some utility functions to generate nice plots and get summary statistics ; I feel like we can provide the "raw" metrics for every measure we actually make and either provide utility functions that do the analysis of these results, or even incorporate those directly into the process and directly return both the raw data and the summary data.

Reply:
I think this is addressed by the `galaxy_summary`